src/Query/DateParser.ts

```diff
@@ -31,93 +30,74 @@ export class DateParser {
 
         for (const parser of dateRangeParsers) {
             const parsedDateRange = parser(input);
-            if (parsedDateRange !== undefined) {
+            if (parsedDateRange.isValid()) {
```

Comment:
I am not sure that this is a pure refactoring either.

Undefined and not valid seem to me to be quite different concepts.

I think we could possibly resolve this kind of thing by taking a step back (together) and writing user-facing documentation for `DateRange`, that explains the intention and meaning of its behaviour...

Reply:
well this depends where the "user" sits. If the user is using a `parser, then, yes, it is a change of behaviour. But I considered the user to use `parseDateRange(...)`, which returns `DateRange` (previously `[..., ...]`) which could be valid or not. So the `undefined` never leaves the `DateParser` class ie it is never `return`ed outside.

Reply:
> and writing user-facing documentation for DateRange, that explains the intention and meaning of its behaviour...

By user, I meant a programmer working on the Tasks plugin, and re-using `DateRange` in another location.

Or someone reading code that calls methods on `DateRange`

More to talk about, later on! 😄